docs/sdk/src/reference_docs/frame_pallet_coupling.rs

```diff
@@ -0,0 +1,262 @@
+//! same runtime. That is, `A` is expressing that can only work if `B` is present.
+//!
+//! Contrary, when pallets are loosely coupled, `A` expresses that some functionality, expressed via
+//! a trait `F`, needs to be fulfilled. This trait is then implemented by `B`, and the two pallets
+//! are linked together at the runtime level. This means that `A` only relies on the implementation
+//! of `F`, which may be `B`, or another implementation of `F`.
+//!
+//! ## Example
+//!
+//! Consider the following example, in which `pallet-foo` needs another pallet to provide the block
+//! author to it, and `pallet-author` which has access to this information.
+#![doc = docify::embed!("./src/reference_docs/frame_pallet_coupling.rs", pallet_foo)]
+#![doc = docify::embed!("./src/reference_docs/frame_pallet_coupling.rs", pallet_author)]
+//!
+//! ### Tight Coupling Pallets
+//!
+//! To tightly couple `pallet-foo` and `pallet-author`, we use Rust's supertrait system. When a
+//! pallet makes its own `trait Config` be bounded by another pallet's `trait Config`, it is
+//! expressing two things:
+//!
+//! 1. that it can only exist in a runtime if the other pallet is also present.
+//! 2. that it can use the other pallet's functionality.
+//!
+//! `pallet-foo`'s `Config` would then look like:
+#![doc = docify::embed!("./src/reference_docs/frame_pallet_coupling.rs", tight_config)]
+//!
+//! And `pallet-foo` can use the method exposed by `pallet_author::Pallet` directly:
+#![doc = docify::embed!("./src/reference_docs/frame_pallet_coupling.rs", tight_usage)]
+//!
+//!
+//! ### Loosely  Coupling Pallets
+//!
+//! If `pallet-foo` wants to *not* rely on `pallet-author` directly, it can leverage its
+//! `Config`'s associated types. First, we need a trait to express the functionality that
+//! `pallet-foo` wants to obtain:
+#![doc = docify::embed!("./src/reference_docs/frame_pallet_coupling.rs", AuthorProvider)]
+//!
+//! > We sometimes refer to such traits that help two pallets interact as "glue traits".
+//!
+//! Next, `pallet-foo` states that it needs this trait to be provided to it, at the runtime level,
+//! via an associated type:
+#![doc = docify::embed!("./src/reference_docs/frame_pallet_coupling.rs", loose_config)]
+//!
+//! Then, `pallet-foo` can use this trait to obtain the block author, without knowing where it comes
+//! from:
+#![doc = docify::embed!("./src/reference_docs/frame_pallet_coupling.rs", loose_usage)]
+//!
+//! Then, if `pallet-author` implements this glue-trait:
+#![doc = docify::embed!("./src/reference_docs/frame_pallet_coupling.rs", pallet_author_provider)]
+//!
+//! And upon the creation of the runtime, the two pallets are linked together as such:
+#![doc = docify::embed!("./src/reference_docs/frame_pallet_coupling.rs", runtime_author_provider)]
+//!
+//! Crucially, when using loose coupling, we gain the flexibility of providing different
+//! implementations of `AuthorProvider`, such that different users of a `pallet-foo` can use
```

Comment:
or providing no-op implementation, very common in our domain

Reply:
Done.